traffic_ops/traffic_ops_golang/deliveryservice/keys.go

```diff
@@ -136,6 +136,39 @@ func AddSSLKeys(w http.ResponseWriter, r *http.Request) {
 	api.WriteResp(w, r, "Successfully added ssl keys for "+*req.DeliveryService)
 }
 
+// GetSSlKeyExpirationInformation gets expiration information for all SSL certificates.
+func GetSSlKeyExpirationInformation(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+	if !inf.Config.TrafficVaultEnabled {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting SSL keys expiration information from Traffic Vault: Traffic Vault is not configured"))
+		return
+	}
+
+	daysParam := inf.Params["days"]
+	if daysParam != "" {
+		if _, err := strconv.Atoi(daysParam); err != nil {
+			api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, errors.New("days parameter must be an integer"), nil)
+			return
+		}
+	}
+
+	expirationInfos, ok, err := inf.Vault.GetExpirationInformation(inf.Tx.Tx, r.Context(), daysParam)
+	if err != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting SSL keys expiration information from Traffic Vault: "+err.Error()))
+		return
+	}
+	if !ok {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting SSL keys expiration information from Traffic Vault: "))
+		return
+	}
+	api.WriteResp(w, r, expirationInfos)
```

Comment:
The type of the response for this is `null | [ExpirationInfo, ...ExpirationInfo[]]`, I think it'd be better if `GetExpirationInformation` returned empty lists instead of `nil` and then we'd just have `ExpirationInfo[]`